### DISCUSSION TITLE
citar-insert-citation in org-modern prettified tables fails

### DISCUSSION THREAD

**Discussion by nicolas-graves** [minad/org-modern#150] [Q&A] [answered]
It fails with "Cannot insert citation here", while it doesn't fail if `org-modern-table` is nil. 

Here's more context about the piece of citar code which fails : 
```
(if (org-cite--allowed-p context)
          (insert
           (format "[cite%s:%s]" (or style "")
                   (mapconcat (lambda (key) (concat "@" key)) keys "; ")))
        (user-error "Cannot insert a citation here"))
```

**Comment by minad** [comment]
Hi, thanks for the report. I fail to reproduce this with a basic test. I created a table where I then execute `(org-cite--allowed-p (org-element-context))` in a table cell via `M-:`. The return value is `t` with and without `org-modern-mode`. Can you please try to narrow this down a bit more?

**Reply by nicolas-graves** [reply]
I cannot reproduce it anymore in the table I had previously but can still reproduce it with a new table, `org-modern-mode`, `(org-cite--allowed-p (org-element-context))` returns `nil`. 
Cannot reproduce either in `emacs -Q` in a new file. Don't really know where it might be wrong, I'll investigate.

**Reply by nicolas-graves** [reply]
I also use evil, maybe an interference with the normal mode

**Reply by nicolas-graves** [reply]
Doesn't seem to be evil either.

**Reply by nicolas-graves** [reply, marked as answer]
It's only the dashed line that has `(org-cite--allowed-p (org-element-context))` `nil`. 

Probably just a wander where I forgot the dashed line is hidden under the visible line in org-modern, and tried to insert on it. Still find it unintuitive on use since we don't see that space as a line, but that is not an issue on your side.

**Reply by nicolas-graves** [reply]
Just as a suggestion : maybe make the line a tiny bit thicker and a tiny bit less up, so that by having your cursor on the line, you see a conflict between the cursor and the line, but keep the same graphical feel. Right now the top of my cursor just touches the bottom of the line, which may give the impression that you can edit there.

**Reply by minad** [reply]
Which line and how? Can you please propose the necessary code change?

**Reply by nicolas-graves** [reply]
It's the horizontal line, but I don't have the same space under it as in https://raw.githubusercontent.com/minad/org-modern/screenshots/example.gif so it might come from somewhere in my config.

**Reply by minad** [reply]
I am not sure, but the horizontal line cannot easily be made thicker.
